src/jdk.compiler/share/classes/com/sun/source/doctree/RawTextTree.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.source.doctree;
+
+/**
+ * A tree node for a fragment of uninterpreted raw text content.
+ *
+ * <p>
+ * The content may contain any text except that for
+ * {@linkplain InlineTagTree inline tags}.
+ *
+ * <p>The format of the content is indicated by the {@linkplain #getKind() kind}
+ * of the tree node.
+ *
+ * @apiNote
+ * This class may be used to represent tree nodes containing
+ * {@linkplain DocTree.Kind#MARKDOWN Markdown} text.
```

Comment:
Yes, the thinking was to allow it to be open-ended.

In this case, while I think we should specify the ability to use `instanceof`-style checks with this API, I do not think it would be necessary ... there is no need for a subtype for `MarkdownTree`, nor would I expect there to be one for any other kind of raw text.